c10/core/impl/PyInterpreter.cpp

```diff
@@ -88,7 +89,44 @@ void PyInterpreter::disarm() noexcept {
   sizes_fn_ = &noop_sizes_fn;
   sym_sizes_fn_ = &noop_sym_sizes_fn;
   layout_fn_ = &noop_layout_fn;
+  trace_cuda_functions->disarm();
 }
 
+// Defined out-of-line because it needs access to the definition of TensorImpl.
+__ubsan_ignore_function__ c10::intrusive_ptr<TensorImpl> PyInterpreter::detach(
+    const TensorImpl* self) const {
+  return (*detach_fn_)(this, self);
+}
```

Comment:
I did this in my commit because somehow I didn't manage to compile without it, but somehow this appears to be working fine in the CI for the main branch, so I'm wondering if there was something wrong with my setup, but I didn't go to the bottom of it. If you have time can you un-do this and see whether it works for you?

Reply:
I've tried undoing it but unfortunately it still causes compile errors. I can try to investigate further.

Reply:
No worries, doing it this way is reasonable to me, and I'm honestly surprised it was working before, so if you're seeing the same errors let's just leave it here.